controllers/openstackdataplane_controller.go

```diff
@@ -153,44 +141,45 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 			labels := client.MatchingLabels(labelSelector)
 			listOpts = append(listOpts, labels)
 		}
-		err = r.Client.List(ctx, roles, listOpts...)
+
+		err = r.Client.List(ctx, nodeSets, listOpts...)
 		if err != nil {
 			return ctrl.Result{}, err
 		}
-		r.Log.Info("found roles", "total", len(roles.Items))
-		if len(instance.Spec.Roles) > len(roles.Items) {
-			shouldRequeue = true
-		}
 
-		for _, role := range roles.Items {
-			logger.Info("DataPlane deploy", "role.Name", role.Name)
-			if role.Spec.DataPlane != instance.Name {
-				err = fmt.Errorf("role %s: role.DataPlane does not match with role.Label", role.Name)
-				deployErrors = append(deployErrors, "role.Name: "+role.Name+" error: "+err.Error())
+		for _, nodeSet := range nodeSets.Items {
+			logger.Info("DataPlane deploy", "nodeSet.Name", nodeSet.Name)
+			// We don't expect that our nodeSet list will be overly large. As such, linear search is likely
+			// to be computationally more efficient compared to importing and using slices.Contains() here.
+			nodeSetFound := false
+			for _, nodeSetName := range instance.Spec.NodeSets {
+				if nodeSetName == nodeSet.Name {
+					nodeSetFound = true
+				}
 			}
-			r.Log.Info("Role", "DeployStrategy.Deploy", role.Spec.DeployStrategy.Deploy, "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
-			if !role.Spec.DeployStrategy.Deploy {
-				_, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), &role, func() error {
-					r.Log.Info("Reconciling Role", "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
-					helper.GetLogger().Info("CreateOrPatch Role.DeployStrategy.Deploy", "Role.Namespace", instance.Namespace, "Role.Name", role.Name)
-					role.Spec.DeployStrategy.Deploy = instance.Spec.DeployStrategy.Deploy
-					if err != nil {
-						deployErrors = append(deployErrors, "role.Name: "+role.Name+" error: "+err.Error())
-					}
-					return nil
-				})
+			// If the specified nodeSet is found. Then we need to add our label to the nodeSet object
+			// to trigger the Ansible deployment
```

Comment:
Just the label triggers a deployment?

Reply:
I left the `deployStrategy.deploy = true` on the `OpenStackDataPlane` so that it can be the object that defines whether or not we deploy. 

Do we want it on both layers?

Just using the label isn't the nicest at the moment though. I'm open to using something different, but I used the `OpenStackDataPlane` controller to label the `OpenStackDataPlaneNodeSet` once it has found it and trigger the deployment. If `deployStrategy.deploy` in `OpenStackDataPlane` is false, it would never get to add the label. But I can revert the deployStrategy here though if you just want to leave it as is for now and we'll just have it in two places.